Spatialite label rendering

RoProducts opened the pull request:
Implementation for #23 

The rendering is managed from the result styles files, which define rules. Rules contain a filter and a symbolizer. The property "labelName" was added to the filter object to define which database label is additionally read to be used to render the labels.
The properties of the symbolizer "textSize" and "textStrokeColor" are used to set-up the rendering of the label.

The filter is added to a database query which results in a "GeometryIterator". This class is copied as "LabelGeometryIterator" to the map module and extends the original by reading also a second column of the query statement, if available. The data, wrapped as object, is added as the geometries "userdata" object and can be used during rendering. See map/src/main/java/it/geosolutions/android/map/overlay/Shapes.java
If the current zoom level is >= the minimum for spatialite label rendering, the data read from the database is rendered on the path using objects toString() method.

Gnafu commented:
In the symbolizer there is also the "textfield" property, is there a particular reason why you didn't use it?

I haven't tested this pull request yet but I have a couple of questions:
- What happens if the style defines a "labelName" that does not exists in the database? (Example: a simple typo in the style file)
- What if I do not want to filter the data, but just display the labels? With this implementation the "labelName" is in the "filter" object, and a filter needs "PropertyName" , "LowerBoundary" and "UpperBoundary" to be defined.

RoProducts commented:
> In the symbolizer there is also the "textfield" property, is there a particular reason why you didn't use it?
Well first if all, I think this property is used in some other place, isn't it ? I won't touch anything existing. Furthermore it is necessary to access the db, which is done by the filter object. The symbolizer is used to configure the rendering.

> What happens if the style defines a "labelName" that does not exists in the database? (Example: a simple typo in the style file)
The query will fail. You should be careful to provide only correct arguments. The test will fail in this case.

> What if I do not want to filter the data, but just display the labels? With this implementation the "labelName" is in the "filter" object, and a filter needs "PropertyName" , "LowerBoundary" and "UpperBoundary" to be defined.

>You can omit the other properties. Of course one could add another object to a rule object. I agree that I'm not applying a filter here, but adding something to the query.
How would you like to have this structured ?

Gnafu commented:
I confirm that the "textfield" in the symbolizer is not currently used and can be used to define the label field.

The database is queried by the SpatialiteDataSourceHandler and the filter is set in the SpatialiteRenderer:
https://github.com/geosolutions-it/SIIGMobile/blob/master/map/src/main/java/it/geosolutions/android/map/spatialite/renderer/SpatialiteRenderer.java#L138
I suggest to set a different property for the label, leaving the filter only for filtering.

If the user misconfigure the label field, only the label should fail to render: the resulting feature will simply have no label.

Apart from the above-mentioned issues, I tested the rendering with the provided styles and it looks good to me.

RoProducts commented:
Updated. I removed the property from filter and use the textfield now.

I added a check if the column defined by the textfield exists in the database. Only in this case it is added to the query, hence a misspelled name results in your desired rendering without labels.

Gnafu commented:
I had to fix the SpatialiteRenderer: it was not resetting the "AdditionalQueryColumn" when the label field did not exists.
